programmers/다트-게임.js

```diff
@@ -0,0 +1,26 @@
+function solution(dartResult) {
+    const answer = [];
+    let splitResult = dartResult.match(/\d{1,2}[SDT]{1}[*|#]?/g );
+    let num = 0;
+    let bonus = 0;
+    let option = 0;
+    
+    for (let i = 0; i < 3; i++) {
+        num = splitResult[i].match(/\d{1,2}/g);
+        bonus = splitResult[i].match(/[SDT]{1}/g);
+        option = splitResult[i].match(/[*|#]/g);
+        
+        if (bonus == "S")  num = parseInt(num);
+        if (bonus == "D")  num **= 2;
+        if (bonus == "T")  num **= 3;
+        
+        if (option == "*") {
+            num *= 2; 
+            if(i != 0) { answer[i - 1] = answer[i - 1] * 2 }; 
+        }
+        else if (option == "#") num = num * (- 1);
```

Comment:
한 줄로 쓰는 if문은 { } 는 빼도돼 그리고 a = a * 2 는 a *= 2로 표현 가능
```suggestion
        if (option == "*") {
            num *= 2; //이런건 잘되있는데 밑에 같은 경우도 잘 수정해줘
            if (i != 0) answer[i - 1] *= 2 
        }
        else if (option == "#") num *= -1;
```